src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/MicrosoftExtensionsLogging/MicrosoftLoggingWrapper.cs

```diff
@@ -33,16 +33,24 @@ public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall ins
 
         private void RecordLogMessage(MethodCall methodCall, IAgent agent)
         {
-            // MSE Logging doesn't have a timestamp for us to pull so we fudge it here.
-            Func<object, DateTime> getTimestampFunc = mc => DateTime.UtcNow;
+            // We need to manually check if each log message is enabled since our MEL instrumentation takes place before
+            // logs have been filtered to enabled levels
+            var melLoggerInstance = (MEL.ILogger)methodCall.InvocationTarget;
```

Comment:
Is it worth putting this in a try/catch? If we fail to cast for some reason (unlikely, but say they add a new log level), we probably want to handle that?

Reply:
Great question! The common pattern is to not hide exceptions like this as they would occur every time this instrumentation is invoked with potentially large overhead (exceptions aren't lightweight). The instrumentation wrapper framework has top level exception handlers that will disable instrumentation if an exception threshold is reached.

Reply:
Oh that's good failsafe to have, and you're right, exceptions aren't cheap. We could use `as` and `Enum.TryParse()` here, though if you just want it to fail outright so it can be addressed, that's valid, too.

Reply:
If this does fail, I think the logged exception for failed type conversions would be more useful in logs (though the same could be achieved with `as`). I do see we are casting to ILogger twice though, so will only do that once.